array_schema_evolution.go

```diff
@@ -0,0 +1,111 @@
+//go:build experimental
+// +build experimental
+
+/**
+ * @file   array_schema_experimental.go
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares Go bindings for experimental features in TileDB.
+ * Experimental APIs to do not fall under the API compatibility guarantees and
+ * might change between TileDB versions
+ */
+
+package tiledb
+
+/*
+#cgo LDFLAGS: -ltiledb
+#cgo linux LDFLAGS: -ldl
+#include <tiledb/tiledb_experimental.h>
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"runtime"
+	"unsafe"
+)
+
+type ArraySchemaEvolution struct {
+	tiledbArraySchemaEvolution *C.tiledb_array_schema_evolution_t
+	context                    *Context
+}
+
+// NewArraySchemaEvolution creates a TileDB schema evolution object.
+func NewArraySchemaEvolution(tdbCtx *Context) (*ArraySchemaEvolution, error) {
+	arraySchemaEvolution := ArraySchemaEvolution{context: tdbCtx}
+	ret := C.tiledb_array_schema_evolution_alloc(
+		arraySchemaEvolution.context.tiledbContext,
+		&arraySchemaEvolution.tiledbArraySchemaEvolution)
+	if ret != C.TILEDB_OK {
+		return nil, fmt.Errorf("Error creating tiledb arraySchemaEvolution: %s",
+			arraySchemaEvolution.context.LastError())
+	}
+
+	// Set finalizer for free C pointer on gc
+	runtime.SetFinalizer(&arraySchemaEvolution,
+		func(arraySchemaEvolution *ArraySchemaEvolution) {
+			arraySchemaEvolution.Free()
+		})
+
+	return &arraySchemaEvolution, nil
+}
+
+// Free destroys an array schema evolution, freeing associated memory
```

Comment:
I would copy the text from the other `Free` functions which has additional explanation.